android/app/src/main/java/com/salesforce/samples/smartsyncexplorerreactnative/MainApplication.java

```diff
@@ -46,7 +46,7 @@
 
 	private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
 		@Override
-		protected boolean getUseDeveloperSupport() {
+		public boolean getUseDeveloperSupport() {
```

Comment:
So it needs to be changed in the [template](https://github.com/forcedotcom/SalesforceMobileSDK-Templates/blob/master/ReactNativeTemplate/android/app/src/main/java/com/salesforce/reactnativetemplate/MainApplication.java#L49) also